### PR TITLE
introduce MCH ROFRecord.h + modify tracking workflows to process several ROF per TF

### PR DESCRIPTION
**Pull request by pillot**
These modifications allows to send and process several events per time frame. Each individual interaction in the time frame is identified by an MCH ROF record, which contains the interaction record and the location of the associated objects (e.g. tracks) in the overall list. For the moment I only modified the tracking devices.

### CODE REVIEW

File: Detectors/MUON/MCH/Workflow/src/ClusterSamplerSpec.cxx [hunk 1]
```diff
@@ -66,40 +75,77 @@ class ClusterSamplerTask
   //_________________________________________________________________________________________________
   void run(framework::ProcessingContext& pc)
   {
-    /// send the clusters of the current event
+    /// send the clusters of the next events in the current TF
+
+    static uint32_t event(-1);
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Workflow/src/ClusterSamplerSpec.cxx]: unsigned int initialised to negative value ?

**Comment by aphecetche** [comment]
@pillot besides the small comment above this PR is fine. Let me know if you want to merge it "as is" (and you'll fix the small error later on) or if you prefer to fix it before merging.

File: Detectors/MUON/MCH/Workflow/src/ClusterSamplerSpec.cxx (hunk 1 above)
Comment (pillot) [reply Detectors/MUON/MCH/Workflow/src/ClusterSamplerSpec.cxx]: I prefer to fix it now because I don't have plan to re-touch this code soon.

**Comment by pillot** [comment]
Actually I just remember this piece of code was copied in TrackSamplerSpec.cxx. Fixed too.

**Review by aphecetche — APPROVED** [review]

**Pull request merged**